packages/react-components/src/components/VideoGallery.tsx

```diff
@@ -326,10 +341,13 @@ export const VideoGallery = (props: VideoGalleryProps): JSX.Element => {
     ? localScreenShareStreamComponent
     : undefined;
 
-  const videoGalleryLayout =
-    layout === 'floatingLocalVideo' ? (
-      <FloatingLocalVideoLayout
+  let videoGalleryLayout: JSX.Element | undefined = undefined;
+  /* @conditional-compile-remove(pinned-participants) */
+  if (pinnedParticipants && pinnedParticipants.length > 0) {
+    videoGalleryLayout = (
+      <PinnedParticipantsLayout
         remoteParticipants={remoteParticipants}
```

Comment:
A lot of props are shared between the layouts.
Can we have a common memoized object and use it to spread the common props across all layouts.
It will easily shave off atleast 20 lines of code :D

Reply:
Great observation. Refactored.